File: java/infrastructure/src/main/java/dev/ufadevconf/infrastructure/api/GlobalExceptionHandler.java
package dev.ufadevconf.infrastructure.api;

import dev.ufadevconf.application.PickupPointNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import java.util.Map;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(PickupPointNotFoundException.class)
    public ResponseEntity<Map<String, String>> handlePickupPointNotFound(PickupPointNotFoundException exception) {
        return ResponseEntity
                .status(HttpStatus.NOT_FOUND)
                .body(Map.of("error", exception.getMessage()));
    }
}

File: kotlin/infrastructure/src/main/kotlin/dev/ufadevconf/infrastructure/api/GlobalExceptionHandler.kt
package dev.ufadevconf.infrastructure.api

import dev.ufadevconf.application.PickupPointNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(PickupPointNotFoundException::class)
    fun handlePickupPointNotFound(exception: PickupPointNotFoundException): ResponseEntity<Map<String, String>> =
        ResponseEntity
            .status(HttpStatus.NOT_FOUND)
            .body(mapOf("error" to exception.message.orEmpty()))
}
